prometheus/value.go

```diff
@@ -105,10 +118,15 @@ func (v *value) Add(val float64) {
 	}
 
 	for {
-		oldBits := atomic.LoadUint64(&v.valBits)
-		newBits := math.Float64bits(math.Float64frombits(oldBits) + val)
-		if atomic.CompareAndSwapUint64(&v.valBits, oldBits, newBits) {
-			return
+		if n := v.s.Load(); n == nil {
+			v.Set(val)
```

Comment:
`return` needed after this line.

Or just move these two lines outside of the `for` block?